internal/dbsqlc/river_job.sql

```diff
@@ -238,226 +225,42 @@ FROM (
-  FROM
-    river_job
-  WHERE
-    id = @id::bigint
-  FOR UPDATE
-),
-updated_job AS (
-  UPDATE
-    river_job
-  SET
-    finalized_at = @finalized_at::timestamptz,
-    errors = array_append(errors, @error::jsonb),
-    state = 'discarded'::river_job_state
-  FROM
-    job_to_update
-  WHERE
-    river_job.id = job_to_update.id
-    AND river_job.state = 'running'::river_job_state
-  RETURNING river_job.*
-)
-(
-  SELECT
-    river_job.*
-  FROM
-    river_job
-  WHERE
-    river_job.id = @id::bigint
-  UNION
-  SELECT
-    updated_job.*
-  FROM
-    updated_job
-) ORDER BY finalized_at DESC NULLS LAST LIMIT 1;
-
--- name: JobSetErroredIfRunning :one
-WITH job_to_update AS (
-  SELECT
-    id,
-    finalized_at,
-    state
-  FROM
-    river_job
-  WHERE
-    id = @id::bigint
-  FOR UPDATE
-),
-updated_job AS (
-  UPDATE
-    river_job
-  SET
-    scheduled_at = @scheduled_at::timestamptz,
-    errors = array_append(errors, @error::jsonb),
-    state = 'retryable'::river_job_state
-  FROM
-    job_to_update
-  WHERE
-    river_job.id = job_to_update.id
-    AND river_job.state = 'running'::river_job_state
-  RETURNING river_job.*
-)
-(
-  SELECT
-    river_job.*
-  FROM
-    river_job
-  WHERE
-    river_job.id = @id::bigint
-  UNION
-  SELECT
-    updated_job.*
-  FROM
-    updated_job
-) ORDER BY scheduled_at DESC NULLS LAST LIMIT 1;
-
--- name: JobSetSnoozedIfRunning :one
-WITH job_to_update AS (
-  SELECT
-    id,
-    finalized_at,
-    state
-  FROM
-    river_job
-  WHERE
-    id = @id::bigint
-  FOR UPDATE
+    SELECT id
+    FROM river_job
+    WHERE id = @id::bigint
+    FOR UPDATE
 ),
 updated_job AS (
-  UPDATE
-    river_job
-  SET
-    scheduled_at = @scheduled_at::timestamptz,
-    state = 'scheduled'::river_job_state,
-    max_attempts = max_attempts + 1
-  FROM
-    job_to_update
-  WHERE
-    river_job.id = job_to_update.id
-    AND river_job.state = 'running'::river_job_state
-  RETURNING river_job.*
+    UPDATE river_job
+    SET errors       = CASE WHEN @error_do_update::boolean        THEN array_append(errors, @error::jsonb) ELSE errors       END,
+        finalized_at = CASE WHEN @finalized_at_do_update::boolean THEN @finalized_at                       ELSE finalized_at END,
+        max_attempts = CASE WHEN @max_attempts_update::boolean    THEN @max_attempts                       ELSE max_attempts END,
+        scheduled_at = CASE WHEN @scheduled_at_do_update::boolean THEN @scheduled_at                       ELSE scheduled_at END,
+        state = @state
+    FROM job_to_update
+    WHERE river_job.id = job_to_update.id
+        AND river_job.state = 'running'::river_job_state
+    RETURNING river_job.*
 )
-(
-  SELECT
-    river_job.*
-  FROM
-    river_job
-  WHERE
-    river_job.id = @id::bigint
-  UNION
-  SELECT
-    updated_job.*
-  FROM
-    updated_job
-) ORDER BY scheduled_at DESC NULLS LAST LIMIT 1;
-
+SELECT *
+FROM river_job
+WHERE id = @id::bigint
+    AND id NOT IN (SELECT id FROM updated_job)
```

Comment:
(Repost since I tweaked this line and the comment was hidden.)

This is a key change here — by excluding an updated job from the lookup on `river_job`, we can drop the `ORDER BY` + `LIMIT` clauses, which means that when combining all the queries we don't need any `CASE WHEN` shenanigans inside the `ORDER BY` clause.